shell/common/api/electron_api_command_line.cc

```diff
@@ -12,18 +13,40 @@
 
 namespace {
 
-bool HasSwitch(const std::string& name) {
-  return base::CommandLine::ForCurrentProcess()->HasSwitch(name);
+// Chromium will hard CHECK if the switch name is not lowercase.
+bool IsSwitchNameValid(std::string_view switch_name) {
+  return base::ToLowerASCII(switch_name) == switch_name;
 }
```

Comment:
Couplea suggestions here:

- It's better to put the "Switch name must be lowercase" idea here so that the other methods don't have to know about that idea
- We don't need to make a new string to test for lowercase


```suggestion
std::optional<std::string_view> CheckSwitchNameValid(std::string_view key) {
  // Chromium will hard CHECK if the switch name is not lowercase.
  if (!std::ranges::none_of(key, absl::ascii_isupper))
    return "Switch name must be lowercase";

  return {};
 }
```

Reply:
@ckerr I copied that helper from upstream: https://source.chromium.org/chromium/chromium/src/+/main:base/command_line.cc;l=68-70?q=%22bool%20IsSwitchNameValid(st%22&sq=package:chromium

Reply:
Either way is fine. I agree upstream parity is important, regardless of the code :smile_cat: